### PR TITLE
types first

### PR DESCRIPTION
**Pull request by FredTheDino**
Do the type-checking first. This fails for one test, but I decided to carry on instead of getting stuck in details.

### CODE REVIEW

File: sylt-compiler/src/typechecker.rs [hunk 1]
```diff
@@ -981,7 +981,7 @@ impl<'c> TypeChecker<'c> {
                 None
             }
 
-            SK::Use { .. }
+            | SK::Use { .. }
```
Comment (EMattfolk) [inline sylt-compiler/src/typechecker.rs:984]: :+2:

**Review by EMattfolk — APPROVED** [review]

File: sylt-compiler/src/typechecker.rs [hunk 2]
```diff
@@ -1012,7 +1031,7 @@ impl<'c> TypeChecker<'c> {
                 self.namespaces[namespace].insert(ident.name.clone(), Name::Global(Some(name)));
             }
 
-            StatementKind::Definition { ident, kind, ty, value } => {
+            SK::Definition { ident, kind, ty, value } => {
```
Comment (sornas) [inline sylt-compiler/src/typechecker.rs:1034]: :-2:

File: sylt-compiler/src/compiler.rs [hunk 3]
```diff
@@ -514,23 +519,15 @@ impl Compiler {
...
-                blocks: blocks
-                    .into_iter()
-                    .map(|x| Rc::new(RefCell::new(x)))
-                    .collect(),
+        Ok(Prog {
+            blocks: blocks
+                .into_iter()
+                .map(|x| Rc::new(RefCell::new(x)))
+                .collect(),
                 functions: functions.iter().map(|(_, f, _)| *f).collect(),
```
Comment (sornas) [inline sylt-compiler/src/compiler.rs:527]: indentation is off here

**Review by sornas — APPROVED** [review]

File: sylt-compiler/src/typechecker.rs (hunk 2 above)
Comment (FredTheDino) [reply sylt-compiler/src/typechecker.rs:1034]: Konsistency is king!

**Pull request merged**